jetty-http/src/main/java/org/eclipse/jetty/http/ResourceHttpContent.java

```diff
@@ -156,7 +151,26 @@ public ByteBuffer getIndirectBuffer()
             return null;
         try
         {
-            return BufferUtil.toBuffer(_resource, false);
+            ByteBuffer byteBuffer = ByteBufferPoolUtil.resourceToBuffer(_resource, false, _bufferPool);
+            _allocatedBuffers.add(byteBuffer);
```

Comment:
What is the lifecycle of a `ResourceHttpContent`? Can it be put into a cache?  If so, could `getIndirectBuffer` be called again and again, and the list of allocated buffers get longer and longer with release never called or called in the distant future?    Is release always called?

I'd be perhaps more included to have a `ByteBuffer getIndirectBuffer(ByteBufferPool pool)` method that the caller can pass in their pool and then be responsible for doing the release themselves.